app/cdap/components/ConfigurationGroup/index.tsx

```diff
@@ -169,6 +157,16 @@ const ConfigurationGroupView: React.FC<IConfigurationGroupProps> = ({
     onChange(updatedValues);
   }
 
+  // This onUnMount is to make sure we clear out all properties that are hidden.
+  useOnUnmount(() => {
+    const newValues = referenceValueForUnMount.current.values;
+    const configGroups = referenceValueForUnMount.current.configurationGroups;
+    const updatedFilteredValues = removeFilteredProperties(newValues, configGroups);
+    changeParentHandler(updatedFilteredValues);
+  });
+
+  React.useEffect(getOrphanedErrors, [errors]);
```

Comment:
Why does it think these lines are changed?

Reply:
I moved changeParentHandler to be above useOnUnmount